app/models/measure.rb

```diff
@@ -436,4 +426,61 @@ def self.changes_for(depth = 1, conditions = {})
      .limit(TradeTariffBackend.change_count)
      .order(Sequel.desc(:operation_date, nulls: :last))
   end
+
+  def meursing?
+    measure_components.any?(&:meursing?)
+  end
+
+  def zero_mfn?
+    third_country? &&
+      measure_components.count == 1 &&
+      measure_components.first.zero_duty?
+  end
+
+  def expresses_unit?
+    measure_type.expresses_unit? && !ad_valorem?
+  end
+
+  def ad_valorem?
+    ad_valorem_resource?(:measure_components) || ad_valorem_resource?(:measure_conditions)
```

Comment:
```suggestion
    %i[measure_components measure_conditions].any? { |condition| ad_valorem_resource?(condition)}
```
Thoughts?

Reply:
I don't know about this one. I'd probably lean on doing this for 3 or more resources :confused: . Struggling to prefer either implementation, atm

Reply:
👍